### PR TITLE
Linter cleanup of on `pre-commit`

### PR DESCRIPTION
**Pull request by averissimo**
#### Changes description

- Add rule to object names that is CDISC and `ANL` variable friendly
- Convert all `# nolint` to specific rules
- Adds default option to all chunks that will remove any `# nolint` from code in Quarto

Superlinter runs fine on this branch: https://github.com/insightsengineering/tlg-catalog/actions/runs/8111598641/job/22171154040
  - The other linter job always fails on workflow dispatch

Running `lintr::lint_dir(".")` passes without any issues

image

### CODE REVIEW

File: book/test-utils/envir_hook.qmd [hunk 1]
```diff
@@ -38,4 +38,8 @@ knitr::opts_template$set(
     cache = FALSE
   )
 )
+
+knitr::opts_chunk$set(
+  tidy = function(code, ...) gsub(pattern = "#\\s?nolint.*", replacement = "", code)
+)
```
Comment (averissimo) [inline book/test-utils/envir_hook.qmd:44]: Alternatively, a new option could be added and the affected chunks modified accordingly.

```suggestion
  ),
  remove_linter = list(
    tidy = function(code, ...) gsub(pattern = "#\\s?nolint.*", replacement = "", code)
  )
)
```

File: book/graphs/pharmacokinetic/pkpg01.qmd [hunk 2]
```diff
@@ -45,7 +45,7 @@ levels(adpp$ARM) <- c(
 ## Plot with Two Cohorts
 
 ```{r plot1, test = list(plot_v1 = "plot")}
-use_title <- "Plot of Mean (+/- SD) Cummulative Percentage (%) of Recovered Drug in Urine \nby Analyte, Visit: PK Evaluable Patients" # nolint line_length_linter
+use_title <- "Plot of Mean (+/- SD) Cummulative Percentage (%) of Recovered Drug in Urine \nby Analyte, Visit: PK Evaluable Patients" # nolint: line_length. line_length_linter
```
Comment (pawelru) [inline book/graphs/pharmacokinetic/pkpg01.qmd:48]:
```suggestion
use_title <- "Plot of Mean (+/- SD) Cummulative Percentage (%) of Recovered Drug in Urine \nby Analyte, Visit: PK Evaluable Patients" # nolint: line_length_linter.
```

File: book/test-utils/envir_hook.qmd (hunk 1 above)
Comment (pawelru) [reply book/test-utils/envir_hook.qmd:44]: Thanks. I was looking into this and it seems that the suggested way is to have a Lua filter. Please have a look at https://github.com/insightsengineering/tlg-catalog/pull/216 - especially linked discussion

File: book/graphs/pharmacokinetic/pkpg01.qmd (hunk 2 above)
Comment (averissimo) [reply book/graphs/pharmacokinetic/pkpg01.qmd:48]:
```suggestion
use_title <- "Plot of Mean (+/- SD) Cummulative Percentage (%) of Recovered Drug in Urine \nby Analyte, Visit: PK Evaluable Patients" # nolint: line_length.
```

File: book/test-utils/envir_hook.qmd (hunk 1 above)
Comment (averissimo) [reply book/test-utils/envir_hook.qmd:44]: Oh great! I missed that issue/discussion.

I'll revert this then.

```suggestion
  )
)
```

**Review by pawelru — APPROVED** [review]
LGTM 👍 Thank you!

**Pull request merged**